src/main/java/health/ere/ps/retry/Retrier.java

```diff
@@ -0,0 +1,45 @@
+package health.ere.ps.retry;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+
+public class Retrier {
+
+    private Retrier() {
+    }
+
+    public static <T> T callAndRetry(
+        List<Integer> retrySeconds,
+        int retryPeriodMs,
+        RetryAction<T> action,
+        Predicate<T> predicate
+    ) {
+        T result = action.execute();
+        if (predicate.test(result)) {
+            return result;
+        }
+        List<Integer> retries = retrySeconds.stream().filter(Objects::nonNull).sorted().toList();
```

Comment:
isnt this double filter and sorted?

Reply:
It turns out that it is but as general retrier component it must perform filtering and probably sorting